### PR TITLE
HugePages documentation

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Add 1.8 release documentation for feature https://github.com/kubernetes/features/issues/275

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/5419)

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
@sjenning - PTAL

File: _data/tasks.yml [hunk 1]
```diff
@@ -187,6 +187,10 @@ toc:
   section:
   - docs/tasks/manage-gpus/scheduling-gpus.md
 
+- title: Manage HugePages
+  section:
+  - docs/tasks/manage-gpus/scheduling-hugepages.md
```
Comment (sjenning) [inline _data/tasks.yml]: manage-hugepages

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
+---
+approvers:
+- derekwaynecarr
+title: Scheduling HugePages
+---
+
+{% capture overview %}
+
+Kubernetes includes **alpha** support for managing huge pages spread across
```
Comment (sjenning) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: nit: "Kubernetes supports the allocation and consumption of huge pages by applications in a pod as an **alpha** feature."

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+nodes. This page describes how users can consume huge pages and the current
+limitations.
+
+{% endcapture %}
+
+{% capture prerequisites %}
+
+1. Kubernetes nodes must pre-allocate huge pages in order for the node to
+   discover them.  A node may only pre-allocate huge pages for a single size.
```
Comment (sjenning) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: nit: s/discover them/report its huge page capacity/

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+1. A special **alpha** feature gate `HugePages` has to be set to true across the
+   system: `--feature-gates="HugePages=true"`.
+
+The nodes will automatically discover and expose all huge page resources as a
```
Comment (sjenning) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: nit: s/export/report/

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    resources:
+      limits:
+        hugepages-2Mi: 100Mi
+  volumes:
+  - name: hugepage
+    emptyDir:
+      medium: HugePages
+```
+
+- Huge page requests must equal the limits.
```
Comment (sjenning) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: nit: might add "This is the default if limits are specified but requests are not."

**Comment by derekwaynecarr** [comment]
@sjenning - thanks, updated per comments.

**Comment by steveperry-53** [comment]
@derekwaynecarr, We need to have a title in the front matter.

<pre>
---
approvers:
  - derekwaynecarr
title: Manage HugePages
---
</pre>

**Comment by steveperry-53** [comment]
@derekwaynecarr, In scheduling-huge-pages.md, in the front matter, put the title on its own line. Otherwise, the Travis CI check complains.

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
+---
+approvers:
+- derekwaynecarr title: Manage HugePages
+---
+
+{% capture overview %}
+
```
Comment (steveperry-53) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: Immediately after {% capture overview %}, add this line:

{% include feature-state-alpha.md %}

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+Kubernetes supports the allocation and consumption of pre-allocated huge pages
+by applications in a pod as an **alpha** feature.  This page describes how users
+can consume huge pages and the current limitations.
```
Comment (steveperry-53) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: Our style is to capitalize Pod, Container, and other API objects.

"... by applications in a Pod ..."

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+{% endcapture %}
+
+{% capture steps %}
+
+## API
+
+Huge pages can be consumed via container level resource requirements using the
+resource name `hugepages-{size}`, where size is the most compact binary notation
+using integer values supported on a particular node.  For example, if a node
```
Comment (steveperry-53) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: Our style is to use angle brackets for placeholders.

`hugepages-<size>`, where <size> is the ...

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+supports 2048kB page sizes, it will expose a schedulable resource
```
Comment (steveperry-53) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: Should this be 2048 KiB?

File: docs/tasks/manage-hugepages/scheduling-hugepages.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+    emptyDir:
+      medium: HugePages
+```
+
+- Huge page requests must equal the limits.  This is the default if limits are
+  specified, but requests are not.
+- Huge pages are isolated at a pod scope, container isolation is planned in a
+  future iteration.
+- EmptyDir volumes backed by HugePages may not consume more huge page memory
+  than the pod request.
```
Comment (steveperry-53) [inline docs/tasks/manage-hugepages/scheduling-hugepages.md]: HugePages -> huge pages?

**Comment by steveperry-53** [comment]
@derekwaynecarr, A few comments, otherwise lgtm.

**Comment by derekwaynecarr** [comment]
@steveperry-53 @sjenning - updated, this should be good to go.

**Pull request merged**